#10803 update OrderService to return correct bool value for cancel me…

freakphp opened the pull request:
…thod

Update OrderService to return correct bool value for cancel method

### Description
Magento/Sales/Model/Service/OrderService::cancel method was returning always true even if cancel method in Magento/Sales/Model/Order didn't do anything cause of canCancel validation.
OrderService cancel method will invoke canCancel to check if order can be canceled and return correct value.

### Fixed Issues (if relevant)
<!--- Provide a list of fixed issues in the format magento/magento2#<issue_number>, if relevant  -->
1. magento/magento2#10803: When canceling order with OrderService, the cancel method always saves the order and returns true, even if the order can not be canceled

### Manual testing scenarios
<!--- Provide a set of unambiguous steps to test the proposed code change -->
1. Requesting API rest endpoint /V1/orders/:id/cancel for the order which can not be canceled should return false

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

freakphp committed:
#10803 update OrderService to return correct bool value for cancel method

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=10919) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=10919) before we can accept your contribution.<br/>

orlangur commented on app/code/Magento/Sales/Model/Service/OrderService.php:
```diff
@@ -87,7 +87,8 @@ public function __construct(
     {
         $order = $this->orderRepository->get($id);
-        if ((bool)$order->cancel()) {
+        if ((bool)$order->canCancel()) {
```
Please remove `(bool)` casting, it is useless within `if` anyway.

ishakhsuvarov replied:
fixed

ishakhsuvarov committed:
magento/magento2#10919: update OrderService to return correct bool value

 - update OrderService to return correct bool value for cancel method
 - removed redundant type casting
